### PR TITLE
Fix additem/removeitem (bug #2788)

### PR DESCRIPTION
**Pull request by Koncord**
Behavior as the original game.

### CODE REVIEW

File: apps/openmw/mwworld/containerstore.cpp [hunk 1]
```diff
@@ -298,26 +300,20 @@ MWWorld::ContainerStoreIterator MWWorld::ContainerStore::addImp (const Ptr& ptr,
 
     // gold needs special handling: when it is inserted into a container, the base object automatically becomes Gold_001
     // this ensures that gold piles of different sizes stack with each other (also, several scripts rely on Gold_001 for detecting player gold)
-    if (Misc::StringUtils::ciEqual(ptr.getCellRef().getRefId(), "gold_001")
-        || Misc::StringUtils::ciEqual(ptr.getCellRef().getRefId(), "gold_005")
-        || Misc::StringUtils::ciEqual(ptr.getCellRef().getRefId(), "gold_010")
-        || Misc::StringUtils::ciEqual(ptr.getCellRef().getRefId(), "gold_025")
-        || Misc::StringUtils::ciEqual(ptr.getCellRef().getRefId(), "gold_100"))
+    if(MWClass::Miscellaneous().isGold(ptr))
```
Comment (scrawl) [inline apps/openmw/mwworld/containerstore.cpp]: You should use ptr's class `ptr.getClass()` instead of constructing a new one.

**Comment by scrawl** [comment]
Are you sure this won't break non-scripted gold pickup? If I activate an item of `gold_100` that's lying around in the gameworld, that's supposed to add 100 gold pieces to the inventory.

**Comment by Koncord** [comment]
Whoops, I forgot non-scripted pick-up.

File: apps/openmw/mwworld/containerstore.cpp [hunk 2]
```diff
@@ -13,6 +13,8 @@
 #include "../mwmechanics/creaturestats.hpp"
 #include "../mwmechanics/levelledlist.hpp"
 
+#include "../mwclass/misc.hpp"
```
Comment (scrawl) [inline apps/openmw/mwworld/containerstore.cpp]: Include can be removed.

File: apps/openmw/mwscript/containerextensions.cpp [hunk 3]
```diff
@@ -119,6 +131,12 @@ namespace MWScript
                     if (count == 0)
                         return;
 
+                    if(Misc::StringUtils::ciEqual(item, "gold_005")
+                        || Misc::StringUtils::ciEqual(item, "gold_010")
+                        || Misc::StringUtils::ciEqual(item, "gold_025")
+                        || Misc::StringUtils::ciEqual(item, "gold_100"))
+                    item = "gold_001";
```
Comment (scrawl) [inline apps/openmw/mwscript/containerextensions.cpp]: This line looks confusing because of the missing indent.

**Pull request merged**